code/game/objects/items/rogueweapons/shields.dm

```diff
@@ -265,4 +265,78 @@
 			if("onback")
 				return list("shrink" = 0.6,"sx" = 1,"sy" = 4,"nx" = 1,"ny" = 2,"wx" = 3,"wy" = 3,"ex" = 0,"ey" = 2,"nturn" = 0,"sturn" = 0,"wturn" = 0,"eturn" = 0,"nflip" = 8,"sflip" = 0,"wflip" = 0,"eflip" = 0,"northabove" = 1,"southabove" = 0,"eastabove" = 0,"westabove" = 0)
 
+
+/obj/item/rogueweapon/shield/artificer
+	name = "steam shield"
+	desc = "A sturdy wood shield thats been highly modified by an artificer. It seems to have several pipes and gears built into it."
+	icon_state = "artificershield"
+	force = 15
+	throwforce = 10
+	dropshrink = 0.8
+	coverage = 60
+	attacked_sound = list('sound/combat/parry/shield/towershield (1).ogg','sound/combat/parry/shield/towershield (2).ogg','sound/combat/parry/shield/towershield (3).ogg')
+	parrysound = list('sound/combat/parry/shield/towershield (1).ogg','sound/combat/parry/shield/towershield (2).ogg','sound/combat/parry/shield/towershield (3).ogg')
+	max_integrity = 200
+	var/smoke_path = /obj/effect/particle_effect/smoke
+	var/cooldowny
+	var/cdtime = 30 SECONDS
+
+/obj/item/rogueweapon/shield/artificer/attack_self(mob/user)
+	if(cooldowny)
+		if(world.time < cooldowny + cdtime)
+			to_chat(user, span_warning("[src] hisses weakly, It's still building up steam!"))
+			return
+	if(prob(25))
+		smoke_path = /obj/effect/particle_effect/smoke/bad
+	else
+		smoke_path = /obj/effect/particle_effect/smoke
+	var/list/thrownatoms = list()
+	var/atom/throwtarget
+	var/distfromcaster
+	user.visible_message(span_notice("Loud whizzing clockwork and the hiss of steam comes from within [src]."))
+	to_chat(user, span_warning("[user] activates a mechanism on [src]!"))
+	sleep(15)
+	playsound(user, 'sound/items/steamrelease.ogg', 100, FALSE, -1)
+	cooldowny = world.time
+	addtimer(CALLBACK(src,PROC_REF(steamready), user), cdtime)
+	for(var/turf/T in view(1, user))
+		for(var/atom/movable/AM in T)
+			thrownatoms += AM
+	for(var/turf/T in view(2, user))
+		if(T == user.loc)
+			continue
```

Comment:
```suggestion
	for(var/turf/T in oview(2, user))
```
`oview()` is `view()` but without the center turf, which in this case would be `user.loc`.